### PR TITLE
Migration guide for stable

### PR DESCRIPTION
**Pull request by alexandrebodin**
#### Description of what you did:

Add Migration guide for stable. Will rename the folders etc once the guide is done

### CODE REVIEW

File: docs/3.0.0-beta.x/migration-guide/migration-guide-beta.20-to-3.0.0.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+  //...
+  url: `https://domain.com:1337/my-strapi-api`,
+};
+```
+
+::: warning
+Adding a sub path to the url doesn't mean your api is going to be prefixed. You will need to host your app behind a proxy and remove the prefix so strapi receives request like if they where made on the root `/` path.
+:::
+
+You can see this option in action in the following [deployment guide](../deployment/nginx-proxy.md).
```
Comment (derrickmehaffy) [inline docs/3.0.0-beta.x/migration-guide/migration-guide-beta.20-to-3.0.0.md:206]: Change this to point to the higher level guide as there is more options than just Nginx (Caddy, HAProxy, and soon I'll have a guide for Apache)

File: docs/3.0.0-beta.x/migration-guide/migration-guide-beta.20-to-3.0.0.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+```js
+module.exports = {
+  //...
+  admin: {
+    url: '/dashboard',
+  },
+};
+```
+
+You can see this option in action in the following [deployment guide](../deployment/nginx-proxy.md).
```
Comment (derrickmehaffy) [inline docs/3.0.0-beta.x/migration-guide/migration-guide-beta.20-to-3.0.0.md:239]: Same here

File: docs/3.0.0-beta.x/plugins/upload.md [hunk 3]
```diff
@@ -310,20 +310,22 @@ or
 $ yarn add strapi-provider-upload-aws-s3
 ```
 
-To enable the provider, create or edit the file at `./extensions/upload/config/settings.json`
+To enable the provider, create or edit the file at `./config/plugins.js`
```
Comment (derrickmehaffy) [inline docs/3.0.0-beta.x/plugins/upload.md:313]: This is plugins now and not using extensions?

File: packages/strapi/lib/core/load-middlewares.js [hunk 4]
```diff
@@ -17,18 +17,16 @@ module.exports = async function(strapi) {
...
-    loaders.loadLocalMiddlewares(appPath, middlewares),
-    // plugins middlewares
-    loaders.loadPluginsMiddlewares(installedPlugins, middlewares),
-    // local plugin middlewares
-    loaders.loadLocalPluginsMiddlewares(appPath, middlewares),
-  ]);
+  // load installed middlewares
+  await loaders.loadMiddlewareDependencies(installedMiddlewares, middlewares);
+  // internal middlewares
+  await loaders.loadInternalMiddlexares(middlewares);
```
Comment (petersg83) [inline packages/strapi/lib/core/load-middlewares.js:23]: Since we modify this code, can you fix the typo please?

Comment (alexandrebodin) [reply packages/strapi/lib/core/load-middlewares.js:23]: @petersg83 I fixed in the RBAC PR I won't do it here to avoid conflicts later on :)

**Pull request merged**